src/test/scala/fi/oph/kouta/mocks/serviceMocks.scala

```diff
@@ -23,9 +22,10 @@ sealed trait ServiceMocks extends Logging {
   var mockServer:Option[ClientAndServer] = None
   var urlProperties:Option[OphProperties] = None
 
-  def startServiceMocking(port:Int = PortChecker.findFreeLocalPort) = {
+  def startServiceMocking() = {
+    mockServer = Some(startClientAndServer())
+    val port = mockServer.get.getLocalPort
```

Comment:
MockServerin koodin perusteella se vissiin siis etsii aina randomin ja varmasti vapaan portin?

Reply:
Kun pyynnön tekee näin, sen pitäisi pyytää käyttöjärjestelmältä suoraan vapaata porttia, jolloin toi pyyntö ei voi epäonnistua. Sain tästä joitakin tälläisiä virheitä Traviksessa:
```
20191204 17:10:49.024  INFO ContextHandler [ScalaTest-main-running-DiscoverySuite] Started o.e.j.s.ServletContextHandler@3321291a{/,file:///home/travis/build/Opetushallitus/src/main/webapp,AVAILABLE}
fi.oph.kouta.security.AuthorizationServiceSpec *** ABORTED ***
  java.lang.RuntimeException: Exception while binding MockServer to port 53302
  at org.mockserver.lifecycle.LifeCycle.bindPorts(LifeCycle.java:165)
  at org.mockserver.lifecycle.LifeCycle.bindServerPorts(LifeCycle.java:123)
  at org.mockserver.mockserver.MockServer.createServerBootstrap(MockServer.java:100)
  at org.mockserver.mockserver.MockServer.<init>(MockServer.java:44)
  at org.mockserver.mockserver.MockServer.<init>(MockServer.java:34)
  at org.mockserver.integration.ClientAndServer.<init>(ClientAndServer.java:19)
  at org.mockserver.integration.ClientAndServer.startClientAndServer(ClientAndServer.java:30)
  at fi.oph.kouta.mocks.ServiceMocks.startServiceMocking(serviceMocks.scala:27)
  at fi.oph.kouta.mocks.ServiceMocks.startServiceMocking$(serviceMocks.scala:25)
  at fi.oph.kouta.security.AuthorizationServiceSpec.startServiceMocking(AuthorizationServiceSpec.scala:12)
  ...
  Cause: java.net.BindException: Address already in use
  at sun.nio.ch.Net.bind0(Native Method)
  at sun.nio.ch.Net.bind(Net.java:433)
  at sun.nio.ch.Net.bind(Net.java:425)
  at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
  at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:130)
  at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:562)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1358)
  at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:501)
  at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:486)
  at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:1019)
  ...
```